src/main/java/com/gradle/CommonCustomUserDataGradlePlugin.java

```diff
@@ -43,38 +43,47 @@ public void apply(Object target) {
         }
     }
 
-    private void applySettingsPlugin(Settings settings) {
-        settings.getPluginManager().withPlugin("com.gradle.enterprise", __ -> {
-            CustomGradleEnterpriseConfig customGradleEnterpriseConfig = new CustomGradleEnterpriseConfig();
-
-            Object extension = settings.getExtensions().getByName("gradleEnterprise");
-            GradleEnterpriseExtension gradleEnterprise = ProxyFactory.createProxy(extension, GradleEnterpriseExtension.class);
-            customGradleEnterpriseConfig.configureGradleEnterprise(gradleEnterprise);
-
-            BuildScanExtension buildScan = gradleEnterprise.getBuildScan();
-            customGradleEnterpriseConfig.configureBuildScanPublishing(buildScan);
-            CustomBuildScanEnhancements buildScanEnhancements = new CustomBuildScanEnhancements(buildScan, providers, settings.getGradle());
-            buildScanEnhancements.apply();
-
-            BuildCacheConfiguration buildCache = settings.getBuildCache();
-            customGradleEnterpriseConfig.configureBuildCache(buildCache);
+    public static void apply(Object gradleEnterprise, ProviderFactory providers, Settings settings) {
+        applySettingsPlugin(
+                gradleEnterprise,
+                providers,
+                settings
+        );
+    }
 
-            // configuration changes applied in this block will override earlier configuration settings,
-            // including those set in the settings.gradle(.kts)
-            Action<Settings> settingsAction = ___ -> {
-                Overrides overrides = new Overrides(providers);
-                overrides.configureGradleEnterprise(gradleEnterprise);
-                overrides.configureBuildCache(buildCache);
-            };
+    private static void applySettingsPlugin(Object gradleEnterpriseExtension, ProviderFactory providers, Settings settings) {
+        GradleEnterpriseExtension gradleEnterprise = ProxyFactory.createProxy(gradleEnterpriseExtension, GradleEnterpriseExtension.class);
+        CustomGradleEnterpriseConfig customGradleEnterpriseConfig = new CustomGradleEnterpriseConfig();
+
+        customGradleEnterpriseConfig.configureGradleEnterprise(gradleEnterprise);
+
+        BuildScanExtension buildScan = gradleEnterprise.getBuildScan();
+        customGradleEnterpriseConfig.configureBuildScanPublishing(buildScan);
+        CustomBuildScanEnhancements buildScanEnhancements = new CustomBuildScanEnhancements(buildScan, providers, settings.getGradle());
+        buildScanEnhancements.apply();
+
+        BuildCacheConfiguration buildCache = settings.getBuildCache();
+        customGradleEnterpriseConfig.configureBuildCache(buildCache);
+
+        // configuration changes applied in this block will override earlier configuration settings,
+        // including those set in the settings.gradle(.kts)
+        Action<Settings> settingsAction = __ -> {
```

Comment:
can we keep the 3 underscores and spacing from before to minimize the diff?

Reply:
yeah, although the 3 underscores are not needed anymore, they don't collide with 2 underscores, as previously.

Reply:
ah, i see, in that case i think its good to change it for consistency.

Reply:
done